releases/release/list.go

```diff
@@ -134,20 +131,40 @@ func (c *List) Lookup(ctx context.Context, req Requirements) (resp *Release, err
 	span.AddField("release.platform", req.Platform)
 	span.AddField("release.arch", req.Arch)
 
-	if raw, ok := c.cache.Get(req); ok {
+	if resp, ok := c.cacheGet(req); ok {
 		span.AddField("cache_hit", 1)
-		resp = raw.(*Release)
 		return resp, err
 	}
 	span.AddField("cache_hit", 0)
 
 	resp, err = c.s3List.Lookup(ctx, req)
 	if err == nil {
-		c.cache.Add(req, resp)
+		c.cacheAdd(req, resp)
 	}
 	return resp, err
 }
 
+func (c *List) cacheGet(req Requirements) (*Release, bool) {
+	c.cacheMU.Lock()
```

Comment:
could this just be a `RLock` / `RUnlock`?

Reply:
I don't think so, as this cache library does expiry during gets (I think).